R/estimate_ascertainment.R

```diff
@@ -0,0 +1,147 @@
+                                   severity_baseline = 0.014,
+                                   burn_in_value = get_default_burn_in(
+                                     epidist
+                                   ),
+                                   smooth_inputs = FALSE,
+                                   smoothing_window = 1,
+                                   correct_for_delays = FALSE,
+                                   max_date = NULL) {
+  # input checking
+  checkmate::assert_data_frame(data)
+  checkmate::assert_names(
+    colnames(data),
+    must.include = c("date", "cases", "deaths")
+  )
+  checkmate::assert_number(
+    severity_baseline,
+    lower = 0.0, upper = 1.0, finite = TRUE
+  )
+  checkmate::assert_integerish(burn_in_value,
+    lower = 1, len = 1L
+  )
+  checkmate::assert_logical(
+    smooth_inputs,
+    len = 1L, any.missing = FALSE, all.missing = FALSE
+  )
+  checkmate::assert_logical(
+    correct_for_delays,
+    len = 1L, any.missing = FALSE, all.missing = FALSE
+  )
+  checkmate::assert_string(max_date, null.ok = TRUE)
+  if (correct_for_delays) {
+    checkmate::assert_class(epidist, "epidist")
+  }
+
+  # match argument for type
+  type <- match.arg(type, several.ok = FALSE)
+
+  # create empty data frame
+  df_severity <- data.frame()
+
+  if (type == "static") {
+    df_severity <- estimate_static(
+      data,
+      epidist = epidist,
+      correct_for_delays = correct_for_delays
+    )
+  } else if (type == "varying") {
+    df_severity <- estimate_time_varying(
+      data,
+      epidist = epidist,
+      smooth_inputs = smooth_inputs,
+      smoothing_window = smoothing_window,
+      burn_in_value = burn_in_value,
+      correct_for_delays = correct_for_delays
+    )
+
+    df_severity <- df_severity[!is.na(df_severity$severity_me), ]
+
+    # collect the severity at the last date, or the date specified by
+    # the user in `max_date`
+    if (!is.null(max_date)) {
+      df_severity <- df_severity[df_severity$date == max_date, ]
+    } else {
+      df_severity <- df_severity[df_severity$date == max(df_severity$date), ]
+    }
+  }
+
+  # data frame for exports, first scale all values and ensure maximum is 1.0
+  df_out <- apply(
+    df_severity[, grepl("severity", colnames(df_severity), fixed = TRUE)],
+    MARGIN = 2,
+    FUN = function(x) {
+      x_ <- severity_baseline / x
+      x_[x_ > 1.0] <- 1.0
+      x_
+    },
+    simplify = FALSE
+  )
+
+  # re-convert to data.frame from list
+  # here, the estimate called "severity_me" translates to "ascertainment_me"
+  # and the estimate "severity_hi" translates to "ascertainment_lo"
+  # TODO: check if this is correct
+  df_out <- as.data.frame(df_out,
+    row.names = NULL,
+    col.names = c("ascertainment_me", "ascertainment_hi", "ascertainment_lo")
+  )
+
+  # return data with columns in correct order
+  df_out[, c("ascertainment_me", "ascertainment_lo", "ascertainment_hi")]
```

Comment:
As far as I can tell, the reordering will have no effect since the column names where created to be precisely in this order.

Reply:
No, the column names are created `c("ascertainment_me", "ascertainment_hi", "ascertainment_lo")`; the reordering flips the order of low and high to make it consistent with other outputs.